BEAM/ViewModels/InspectionViewModel.cs

```diff
@@ -55,18 +79,77 @@ public InspectionViewModel(SequenceViewModel sequenceViewModel, DockingViewModel
     public void OnClose()
     {
         _currentSequenceViewModel.UnregisterInspectionViewModel(this);
+        
+        if (!_analysisRunning) return;
+        AbortAnalysis();
+        ProgressWindow.Close();
     }
 
     /// <summary>
     /// When the user interacted with the view, the coordinates of where the
     /// pointer was pressed and released, are passed to this method.
+    ///
+    /// Does not update, if KeepData is activated or an analysis is already running.
     /// </summary>
     public void Update(Coordinate2D pressedPoint, Coordinate2D releasedPoint)
     {
         if (KeepData) return;
-        _pointerRectanglePosition = (pressedPoint, releasedPoint);
-        Plot result = _currentAnalysis.Analyze(pressedPoint, releasedPoint, _currentSequenceViewModel.Sequence);
-        CurrentPlot = result;
+        if (_analysisRunning)
+        {
+            Models.Log.Logger.GetInstance().Warning(LogEvent.Analysis, "Analysis is already running.");
+            return;
+        }
+        
+        StartAnalysis(pressedPoint, releasedPoint);
+    }
+
+    private void StartAnalysis(Coordinate2D pressedPoint, Coordinate2D releasedPoint)
+    {
+        // Only display progress for more than 100 Pixels analysed.
+        if (AmountPixels(pressedPoint, releasedPoint) > 100)
+        {
+            ProgressWindow = new AnalysisProgressWindow(this);
+            ProgressWindow.Show();
+        }
+        _analysisRunning = true;
+        _currentAnalysis.Analyze(pressedPoint, releasedPoint, _currentSequenceViewModel.Sequence,
+            this, _cancellationTokenSource.Token);
+    }
+
+    /// <summary>
+    /// Calculate the amount of pixels encompassed by the rectangle parallel to the axes with these corners. 
+    /// </summary>
+    /// <param name="pressedPoint"></param>
+    /// <param name="releasedPoint"></param>
+    /// <returns></returns>
+    private static double AmountPixels(Coordinate2D pressedPoint, Coordinate2D releasedPoint)
+    {
+        return Math.Abs((pressedPoint.Column - releasedPoint.Column) * (pressedPoint.Row - releasedPoint.Row));
+    }
+
+    /// <summary>
+    /// Stops the currently running Analysis
+    /// </summary>
+    public void AbortAnalysis()
+    {
+        if (!_analysisRunning) return;
+        _cancellationTokenSource.Cancel();
+        _cancellationTokenSource.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
This may be too fast.
What happens if this thread is faster than the analysis thread? you are instantly creating a new (uncancelled) token, so the analysis continues to be executed.